packages/babel-plugin/src/utils/manipulate-template-literal.ts

```diff
@@ -0,0 +1,148 @@
+import type { Metadata } from '../types';
+
+import { getPathOfNode } from './ast';
+
+const conditionalPaths: ['consequent', 'alternate'] = ['consequent', 'alternate'];
+
+/**
+ * Returns `true` if the CSS property or pseudo classes/pseudo elements
+ * are defined within a template element (the string pieces of the template literal)
+ * rather than in the expression.
+ *
+ * For example
+ * - CSS property: `color: ${({ isPrimary }) => (isPrimary ? 'green' : 'red')};`
+ * - Pseudo class: `hover: ${({ isPrimary }) => (isPrimary ? 'color:green' : 'color:red')};`
+ * - Pseudo element: `:before {${({ isPrimary }) => (isPrimary ? 'color: green' : 'color:red')}};`
+ * @param node
+ */
+export const isCssPropertyInTemplateElement = (node: t.TemplateElement): boolean => {
+  const value = node.value.raw;
+
+  return value.includes(':');
+};
+
+/**
+ * TODO: this is a temporary workaround so that we don't evaluate expressions that may throw an error.
+ * It should be removed after addressing https://github.com/atlassian-labs/compiled/issues/1081
+ *
+ * Returns `true` if
+ * - there is another conditional expression at the root of the current template literal
+ *   with a conditional expression to evaluate OR
+ * - there is a nested logical expression within the current template literal
+ *
+ * @param node
+ * @param meta {Metadata} The current metadata to use to find the parent node
+ */
+export const hasNestedTemplateLiteralsWithConditionalRules = (
+  node: t.TemplateLiteral,
+  meta: Metadata
+): boolean => {
+  const expressionPath = getPathOfNode(node, meta.parentPath);
+  const { parent } = expressionPath;
+  let isNested = false;
+
+  traverse(parent, {
+    noScope: true,
+    ConditionalExpression(path) {
+      conditionalPaths.map((c) => {
+        const expression = path.node[c];
+
+        if (
+          (t.isTaggedTemplateExpression(expression) && expression.quasi === node) ||
+          (t.isTemplateLiteral(expression) &&
+            expression.expressions.some((expressionNode) =>
+              t.isArrowFunctionExpression(expressionNode)
+            )) ||
+          t.isLogicalExpression(expression)
+        ) {
+          isNested = true;
+          path.stop();
+        }
+      });
+    },
+  });
+
+  return isNested;
+};
+
+/**
+ * Manipulates template literal so that it resembles
+ * CSS `property: value` declaration where
+ * - before: all the CSS _before_ the interpolation
+ * - after: all the CSS _after_ the interpolation
+ * @param expression
+ * @param quasi
+ * @param nextQuasis
+ */
+export const moveCssPropertyInExpression = (
+  expression: t.ConditionalExpression,
+  quasi: t.TemplateElement,
+  nextQuasis: t.TemplateElement,
+  deleteValue = true
+): void => {
+  const value = quasi.value.raw;
+  const parts = value.split(';');
+  const before = parts[parts.length - 1]; // everything after the last ';'. Hence, after the last CSS declaration
+  const nextQuasisExtractedCss = cssAfterInterpolation(nextQuasis.value.raw);
```

Comment:
Could you leave `cssAfterInterpolation` as private, and use `cssAffixInterpolation` where before is an empty string?